website/source/docs/configuration/providers.html.md

```diff
@@ -117,3 +117,14 @@ KEY {
 	CONFIG
 }
 ```
+
+## Interpolation
+Providers support [interpolation syntax](/docs/configuration/interpolation.html) allowing dynamic configuration at run time.
+
+```
+provider "aws" {
+	region = "${var.aws_region}"
+}
+```
+
+Only [variables](/docs/configuration/variables) and [remote state](/docs/state/remote.html) are supported, it is not possible to use the output from a resource, module or data source in the interpolation syntax for a provider.
```

Comment:
I have updated the comments to update the description, also ran some tests this morning to confirm that remote state works as I thought.  Not sure if it is worth adding the example?
```hcl
data "terraform_remote_state" "nomad" {
  backend = "s3"

  config {
    bucket  = "nic-terraform-state"
    key     = "examples/nomad/terraform.state"
    region  = "eu-west-1"
    profile = "hashicorp"
  }
}

provider "nomad" {
  address = "${format("http://%s:4646",data.terraform_remote_state.nomad.alb_nomad)}"
}
```